components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -43,7 +45,24 @@ class WebExtensionBrowserMenuBuilder(
         val extensionMenuItems =
             WebExtensionBrowserMenu.getOrUpdateWebExtensionMenuItems(store.state, store.state.selectedTab)
 
-        val webExtMenuItem = if (extensionMenuItems.isNotEmpty()) {
+        val filteredExtensionMenuItems = extensionMenuItems.toMutableList()
+
+        val finalList = items.toMutableList()
+
+        extensionMenuItems.forEach { webExtensionBrowserMenuItem ->
+            val index = finalList.indexOfFirst { browserMenuItem ->
+                (browserMenuItem as? PlaceholderBrowserMenuItem)?.label ==
+                    (webExtensionBrowserMenuItem as? WebExtensionBrowserMenuItem)?.id
+            }
+            // replace placeholders with corresponding web extension, and remove it from extensions menu list
+            if (index != -1) {
+                finalList.removeAt(index)
+                finalList.add(index, webExtensionBrowserMenuItem)
+                filteredExtensionMenuItems.remove(webExtensionBrowserMenuItem)
+            }
```

Comment:
We could write this block is a more succinct way e.g.:

```kotlin
val filteredExtensionMenuItems = extensionMenuItems.filter { webExtensionBrowserMenuItem ->
  // Check if we have a placeholder
  val index = finalList.indexOfFirst { browserMenuItem ->
    (browserMenuItem as? PlaceholderBrowserMenuItem)?.label == webExtensionBrowserMenuItem.id
  }
  // Replace placeholder with corresponding web extension, and remove it from extensions menu list
  if (index != -1) {
    finalList[index] = webExtensionBrowserMenuItem
  }
  index == -1
}
```

This way we can skip one mutable list too. It's also possible to write it without mutable lists, but either way let's add a test to `WebExtensionBrowserMenuBuilderTest` please :).